src/lib/dereferencing.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * This module handles template interpolation.
+ */
+
+import { StepMatcher } from '@/lib/matcher';
+import * as S from '@/lib/steps';
+
+export type PipelinesScopeContext = {
+  [pipelineName: string]: S.Pipeline;
+};
+
+type StepDereferencer = (step: S.PipelineStep) => S.PipelineStep;
+
+/**
+ * The `PipelineDereferencer` class provides a "dereferencing" implementation
+ * for each possible step in a pipeline. It replaces pipeline names (saved in
+ * store) referenced in steps configuration by their corresponding pipelines.
+ *
+ * The `dereference` method takes a step as parameter and returns a new version
+ * of this step, with pipeline names dereferenced.
+ */
+export class PipelineDereferencer implements StepMatcher<S.PipelineStep> {
```

Comment:
Do we really need a a whole class to compute deferencement? It looks like we're killing a fly with a sledgehammer. Wouldn't a simple function like this one be enough?

```typescript
const dereferenced = [];
for (const step of pipeline.steps) {
  let newStep;
  if (step.name === 'append') { newStep = ...; }
  else { newStep = { ...step } };
  dereferenced.push(newStep);
}
```

Reply:
If we do need a whole class, then maybe a "noop" class for the TemplatingClass and this one would be nice to avoid all the "empty" methods that just copy the input step.